jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/CachedContentFactory.java

```diff
@@ -403,7 +404,7 @@ public class CachedHttpContent implements HttpContent
 
             _lastAccessed = Instant.now();
 
-            _etag = CachedContentFactory.this._etags ? new PreEncodedHttpField(HttpHeader.ETAG, resource.getWeakETag()) : null;
+            _etag = CachedContentFactory.this._etags ? new PreEncodedHttpField(HttpHeader.ETAG, EtagUtil.calcWeakEtag(resource.getPath())) : null;
```

Comment:
This would only work with path-backed resources. Shouldn't we be capable of calculating the etag of a potential `MemoryResource`?

Reply:
Fixing.